charger/go-e/api.go

```diff
@@ -0,0 +1,134 @@
+package goe
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/evcc-io/evcc/util"
+	"github.com/evcc-io/evcc/util/request"
+)
+
+const CloudURI = "https://api.go-e.co"
+
+// Response is the v1 and v2 api response interface
+type Response interface {
+	Status() int
+	Enabled() bool
+	CurrentPower() float64
+	ChargedEnergy() float64
+	Currents() (float64, float64, float64)
+	Identify() string
+}
+
+type API interface {
+	Status() (Response, error)
+	Update(payload string) (Response, error)
+}
+
+type LocalAPI struct {
+	*request.Helper
+	uri string
+	v2  bool
+}
+
+func NewLocal(log *util.Logger, uri string) *LocalAPI {
+	uri = strings.TrimRight(uri, "/")
+	uri = strings.TrimSuffix(uri, "/api")
+
+	api := &LocalAPI{
+		Helper: request.NewHelper(log),
+		uri:    uri,
+	}
+
+	if api.IsV2() {
+		api.v2 = true
+		api.uri = api.uri + "/api"
+	}
+
+	return api
+}
+
+func (c *LocalAPI) IsV2() bool {
+	_, err := c.Response("/api/status", "")
+	return err == nil
+}
+
+func (c *LocalAPI) Response(function, payload string) (Response, error) {
+	var status Response
+	if c.v2 {
+		status = &StatusResponse2{}
+	} else {
+		status = &StatusResponse{}
+	}
+
+	url := fmt.Sprintf("%s/%s", c.uri, function)
+	if payload != "" {
+		if c.v2 {
+			url += "set?" + payload
+		} else {
+			url += "?payload=" + payload
+		}
+	}
+
+	err := c.GetJSON(url, &status)
+	return status, err
+}
+
+func (c *LocalAPI) Status() (Response, error) {
+	return c.Response("status", "")
+}
+
+func (c *LocalAPI) Update(payload string) (Response, error) {
+	return c.Response("mqtt", payload)
```

Comment:
Keine Ahnung ob das einen Vor- oder Nachteil bietet das update per mqtt zu machen oder direkt über die v2 API in dem  Fall.
Ich hab es aber noch nicht getestet.

Reply:
Ähhh- das Update hat noch gar keine Fallunterscheidung. Hab ich vergessen mangels Testgerät :)